### ISSUE TITLE
Move iter8 chart to iter8-tools/hub repository

### ISSUE THREAD

**Issue by kalantar** [iter8-tools/iter8#1352]
Today the `iter8` command uses a single `iter8` Helm chart to create experiment definitions. The `iter8` chart is defined in the `iter8-tools/iter8` repository. 

The implementation downloads the chart from the file system assuming it is a GitHub repository. However, it is structured as a Helm chart and treating it as such would simplify the implementation of Iter8. To do this:

1. move the chart to the `iter8-tools/hub` repository; the hub repository is already exported as a Helm repository 
2. modify `iter8` commands to use this Helm repository.

In the process, we want to be sure not to disrupt the development of automated experiments (i.e., `autox`) and we want to continue to support execution using a local chart for chart development use cases.

**Comment by kalantar** [comment]
The current implementation of `iter8 launch` and `iter8 k launch` downloads the chart (assuming it is from a GitHub repository) using the internal implementation of `iter8 hub`), processes the downloaded template using the internal implementation of `iter8 gen` and runs the resulting experiment (file or secret) using the internal implementation of `iter8 run`. 

While something similar is still needed for `iter8 launch`, a more Helm like implementation of `iter8 k launch` could more easily be accomplished using the equivalent of `helm install` and/or `helm upgrade`.

By default, the Helm repository and chart name are defaulted (to `https://iter8-tools.github.io/hub` and `iter8`, respectively). The launch command is then similar to:
```shell
iter8 --set "tasks={http}" --set http.url=http://httpbin.default/get --set runner=job
```

Optionally, the Helm repository and chart can set using options:
- `--repo <helm_repository_url>`  identifies a Helm repository by URL
- `--chartName <chart_name>` identifies a chart name

Furthermore, Helm allows identifying a chart using the form `repo/chart` where `repo` is a name added using the `helm repo add` command.

In all cases where a remote repository is specified, Helm commands offer a set of access options (`--ca-file`, `--cert-file`, `--key-file`, `--insecure-skip-verify`, `--keyring`, `--password`, `--pass-credentials`, `--username` and `--verify`

Finally, to indicate that a chart is local, `--localChart` can be specified.

When charts are pulled using the Helm APIs, there is no need to update the dependencies; the latest dependencies are already pulled.

The same options can be made available to local execution (`iter8 launch`).

In summary, the following changes are to be made to the commands
- `iter8 gen` - remove update dependencies; hide
- `iter8 hub` - remove this command (user can use `helm pull` if desired instead)
- `iter8 launch` - add `ChartPathOptions`, `--localChart` flags; remove `--chartsParentDir`
- `iter8 run` - hide
- `iter8 k launch` - add `ChartPathOptions`, `--localCharat` flags; remove `--chartsParentDir`, `--noDownload`, `--remoteFolderURL`

**Comment by kalantar** [comment]
Done; closing.
